org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -230,6 +232,20 @@ public List<Either<Command, CodeAction>> getSourceActionCommands(CodeActionParam
 		Optional<Either<Command, CodeAction>> generateFinalModifiers = addFinalModifierWherePossibleAction(context);
 		addSourceActionCommand($, params.getContext(), generateFinalModifiers);
 
+		// Add add javadoc quick assist

```

Comment:
Yeah, we should put it in `QuickAssistProcessor`. 

The reason I did put it in `SourceAssistProcessor` is that I'm working on another PR to reorder all the quick assists: https://github.com/eclipse/eclipse.jdt.ls/pull/2109, I'm considering to put "generate javadoc" under "generating accessors", and so on. But never mind, we can also do the reordering for all quick assists in `CodeActionHandler.java`, I will update that PR then.

Besides, we can also consider moving some existing quick assists (e.g., generating accessor, generating constructor) from `SourceAssistProcessor` to `QuickAssistProcessor` for better categorizing, but it will bring some extra calculations.

Reply:
Yup, I noticed this as well in the one about expanding the `final` modifier to more cases. Thanks for handling this.